torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -432,6 +475,9 @@ void TensorPipeAgent::startImpl() {
     }
     std::unique_ptr<TransportRegistration> reg =
         TensorPipeTransportRegistry()->Create(key);
+    if (!reg->transport->isViable()) {
+      continue;
+    }
```

Comment:
This change (and the below, for consistency) is an unforeseen design flaw of how we handle "viability" for backends (which checks for preconditions, like the ability to find and load libibverbs.so in this case). Ideally viability would have been a "concept" that is entirely internal to TensorPipe, without leaking to downstream libraries. However I had missed one aspect, which is that for transports the downstream library needs to collect the addresses, and if it cannot collect the address of a non-viable transport. So the temporary workaround is this extra check, but I'll try to rework our design to be able again to hide this detail.